wgsl/index.bs

```diff
@@ -454,6 +454,149 @@ whether failure to meet a particular requirement
 results in a shader-creation, pipeline-creation, or dynamic error.
 
 The WebGPU specification describes the consequences of each kind of error.
+A WGSL program with a [=shader-creation error=] or [=pipeline-creation error=] error will not be incorporated
+into a [=pipeline=] and hence will not be executed.
+
+## Diagnostics ## {#diagnostics}
+
+An implementation can generate [=diagnostics=] during [=shader module creation=].
+A <dfn noexport>diagnostic</dfn> is a message produced by the implementation for the benefit of the application author.
+
+We say a diagnostic is <dfn noexport>triggered</dfn> when particular condition is met, known as the [=diagnostic/triggering rule=].
+The place in the source text where the condition is met is known as the <dfn dfn-for="diagnostic">triggering location</dfn>.
+
+A [=diagnostic=] has the following properties:
+* A [=diagnostic/severity=].
+* A [=diagnostic/triggering rule=].
+* A [=diagnostic/triggering location=], a source location where the triggering rule applies, expressed as a point or range in the source text.
+* Optionally, details describing how the [=diagnostic/triggering rule=] applies in this case.
+
+The <dfn dfn-for="diagnostic">severity</dfn> of a diagnostic is one of:
+: <dfn dfn-for="severity" noexport>error</a>
+:: The diagnostic an error.
+     This is the greatest severity.
+: <dfn dfn-for="severity" noexport>warning</a>
+:: The diagnostic describes an anomaly that merits the attention of the application developer, but is not an error.
+: <dfn dfn-for="severity" noexport>info</a>
+:: The diagnostic describes a notable condition that merits attention of the application developer, but is not an error or warning.
+    This is the lowest severity.
+
+Triggered diagnostics [=behavioral requirement|will=] be processed as follows:
+1. If a [=shader-creation error=] occurs independently of diagnostic processing, then all diagnostics may be discarded.
+2. Sort [=diagnostic filters=] so that DF1 precedes DF2 if the DF1's [=affected range=] includes DF2's affected range.
+    * Apply the diagnostic filters, in this order, to all diagnostics.
+3. If there is at least one remaining diagnostic with `error` severity, then other diagnostics *may* be discarded.
```

Comment:
This implies that if the user has globally lowered the severity of `derivative_uniformity` to `warning`, we must show all warnings that the analysis would be able to find (which is undesirable from an implementation standpoint). Unless there's wording in the analysis that says it can stop after it finds the first violation, but I couldn't find such wording after a cursory scan.

Reply:
I deliberately made it that way for warnings.  It seemed that was the only reasonable thing, but tell me why I'm wrong.
I deliberately allowed fail-on-first-error.

Reply:
Jim suggested you might be concerned about the combinatorially many warnings generated by a uniformity problem because of the many paths that might go through a silenced node. Is that the concern?

Reply:
Correct. I do not believe it is feasible to require that an implementation must find all uniformity violations that exist in the graph.